code/pages/ExtensibleSearchPage.php

```diff
@@ -719,11 +748,76 @@ public function getSearchResults($data = null, $form = null) {
 
 		$startTime = microtime(true);
 
+		if($results = $this->getSearchEngineResults($data, $form)) {
```

Comment:
That's the thing, full-text *already is* implemented as a search engine type. It's just that the logic in the extensible search page has never been cleaned up, and was noting it as being the "fallback", which was confusing at first glance. The full-text search *should* be encapsulated in the get search engine results function, as stated in my comment above.